lychee-lib/src/collector.rs

```diff
@@ -50,13 +53,27 @@ impl Collector {
         self
     }
 
+    /// Pass a [`BasicAuthExtractor`] which is capable to match found
+    /// URIs to basic auth credentials. These credentials get passed to the
+    /// request in question.
+    #[must_use]
+    #[allow(clippy::missing_const_for_fn)]
+    pub fn basic_auth_extractor(mut self, extractor: BasicAuthExtractor) -> Self {
+        self.basic_auth_extractor = Some(extractor);
+        self
+    }
+
     /// Fetch all unique links from inputs
     /// All relative URLs get prefixed with `base` (if given).
     /// (This can be a directory or a base URL)
     ///
     /// # Errors
     ///
     /// Will return `Err` if links cannot be extracted from an input
+    ///
+    /// # Panics
+    ///
+    /// Will panic if the [`BasicAuthExtractor`] cannot be constructed.
```

Comment:
I guess this isn't needed anymore. We should remove it again.

Reply:
done

Reply:
lgtm!